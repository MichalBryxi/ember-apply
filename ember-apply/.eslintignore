declarations/
node_modules/

src/-private/automation/
