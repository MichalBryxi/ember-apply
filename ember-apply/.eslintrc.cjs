"use strict";

const { configs } = require("@nullvoxpopuli/eslint-configs");

const config = configs.node();

module.exports = {
  ...config,
  overrides: [
    ...config.overrides,
    {
      files: ["tests/**/*.{ts,js}"],
      rules: {
        "n/no-missing-import": ["off"],
      },
    },
  ],
};
