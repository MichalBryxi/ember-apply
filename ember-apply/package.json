{
  "name": "ember-apply",
  "version": "2.10.1",
  "description": "automatic integration and configuration from the community",
  "repository": {
    "url": "https://github.com/NullVoxPopuli/ember-apply",
    "type": "https",
    "directory": "packages/ember-apply"
  },
  "exports": {
    ".": {
      "import": "./src/index.js",
      "types": "./declarations/src/index.d.ts"
    },
    "./test-utils": "./src/test-utils.js"
  },
  "main": "src/index.js",
  "type": "module",
  "author": "NullVoxPopuli",
  "license": "MIT",
  "private": false,
  "bin": {
    "ember-apply": "src/cli/index.js"
  },
  "typesVersions": {
    "*": {
      "*": [
        "./declarations/src/*",
        "./declarations/src/index.d.ts"
      ]
    }
  },
  "types": "./declarations/src/",
  "files": [
    "src/**/*",
    "declarations/**/*",
    "README.md"
  ],
  "scripts": {
    "prepack": "pnpm build && cp ../README.md ./",
    "build": "tsc --build",
    "lint:types": "tsc",
    "lint": "concurrently \"npm:lint:*(!fix)\" --names \"lint:\"",
    "lint:fix": "concurrently \"npm:lint:*:fix\" --names \"fix:\"",
    "lint:js": "eslint .",
    "lint:prettier": "prettier --check .",
    "lint:prettier:fix": "prettier --write .",
    "test": "vitest --coverage --no-watch --no-threads",
    "test:watch": "vitest --watch --no-threads"
  },
  "dependencies": {
    "@babel/core": "^7.23.2",
    "@babel/preset-env": "^7.23.2",
    "chalk": "^5.3.0",
    "ember-template-recast": "^6.1.4",
    "execa": "^8.0.1",
    "find-up": "^6.3.0",
    "fs-extra": "^11.1.1",
    "globby": "^13.2.2",
    "jscodeshift": "^0.15.0",
    "latest-version": "^7.0.0",
    "ora": "^6.3.1",
    "pacote": "^15.2.0",
    "postcss": "^8.4.31",
    "posthtml": "^0.16.5",
    "posthtml-boolean-attributes": "^0.3.1",
    "prettier": "^3.0.3",
    "semver": "^7.5.4",
    "sort-object-keys": "^1.1.3",
    "unified": "^11.0.0",
    "yargs": "^17.7.2",
    "yarn-workspaces-list": "^0.2.0"
  },
  "devDependencies": {
    "@babel/eslint-parser": "^7.22.15",
    "@nullvoxpopuli/eslint-configs": "^3.2.2",
    "@types/fs-extra": "^11.0.2",
    "@types/jscodeshift": "^0.11.7",
    "@types/node": "^20.8.4",
    "@types/pacote": "^11.1.6",
    "@types/semver": "^7.5.3",
    "@types/sort-object-keys": "^1.1.1",
    "@types/yargs": "^17.0.28",
    "@typescript-eslint/eslint-plugin": "^6.7.5",
    "@typescript-eslint/parser": "^6.7.5",
    "@vitest/ui": "^0.34.6",
    "c8": "^8.0.1",
    "concurrently": "^8.2.1",
    "eslint": "^8.51.0",
    "eslint-plugin-decorator-position": "^5.0.2",
    "eslint-plugin-import": "^2.28.1",
    "eslint-plugin-node": "^11.1.0",
    "eslint-plugin-prettier": "^5.0.1",
    "prettier": "^3.0.0",
    "typescript": "5.2.2",
    "vite": "^4.4.11",
    "vitest": "^0.34.6"
  },
  "engines": {
    "node": ">=16.0.0"
  },
  "volta": {
    "extends": "../package.json"
  }
}
