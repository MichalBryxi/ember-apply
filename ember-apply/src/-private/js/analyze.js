// @ts-check
/**
 *
 * @typedef {import('jscodeshift')} JSCodeshift
 * @typedef {ReturnType<JSCodeshift>} jAST
 *
 * @typedef {object} CallbackApi
 * @property {jAST} root
 * @property {JSCodeshift} j
 *
 * @callback TransformCallback
 * @param {CallbackApi} callbackApi
 * @return {Promise<void>} return
 *
 * @typedef {object} Options
 * @property {Parameters<JSCodeshift['withParser']>[0]} [parser]
 *
 */
import fs from 'fs/promises';
import jscodeshift from 'jscodeshift';
import path from 'path';

/**
 * reads a script from source and allows analysis via jscodeshift.
 * This does not write to the file afterwards.
 *
 * The https://astexplorer.net/ REPL can be used to inspect and test out how
 * to transform javascript and typescript files.
 *
 * For javascript, the `babel` parser is used.
 * For typescript, the `ts` parser is used.
 *
 * @example
 * ```js
 * import { js } from 'ember-apply';
 *
 * await js.analyze('path/to/file.js', ({ root, j }) => {
 *   root
 *    .find(j.ImportDeclaration)
 *    .forEach(path => {
 *       // do some analysis on the found AST Nodes
 *     })
 *   });
 * ```
 *
 * @param {string} filePath to the file to transform
 * @param {TransformCallback} callback
 * @param {Options} [options]
 * @returns {Promise<string>} the transformed source
 */
export async function analyze(filePath, callback, options = {}) {
  let code = (await fs.readFile(filePath)).toString();

  let j;

  if (options?.parser) {
    j = jscodeshift.withParser(options.parser);
  } else {
    if (path.extname(filePath).endsWith('ts')) {
      j = jscodeshift.withParser('ts');
    } else {
      j = jscodeshift.withParser('babel');
    }
  }

  let root = j(code);

  await callback({ root, j });

  let transformed = root.toSource();

  return transformed;
}
