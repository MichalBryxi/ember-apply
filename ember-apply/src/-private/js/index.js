// @ts-check

/**
 * Public API for the js namespace
 */

export { analyze } from './analyze.js';
export { transform } from './transform.js';
