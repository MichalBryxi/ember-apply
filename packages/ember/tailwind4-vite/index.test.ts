import { apply, diff, diffSummary, newEmberApp } from "ember-apply/test-utils";
import { describe, expect, it } from "vitest";

import { default as tailwind4Vite } from "./index.js";

describe("tailwind4-vite", () => {
  it("default export exists", () => {
    expect(typeof tailwind4Vite).toEqual("function");
  });

  describe("applying to an ember app", () => {
    it("works via CLI", async () => {
      let appLocation = await newEmberApp([
        "--blueprint",
        "@embroider/app-blueprint",
      ]);

      await apply(appLocation, tailwind4Vite.path);

      expect(await diffSummary(appLocation)).toMatchSnapshot();
      expect(
        await diff(appLocation, { ignoreVersions: true })
      ).toMatchSnapshot();
    });
  });
});
