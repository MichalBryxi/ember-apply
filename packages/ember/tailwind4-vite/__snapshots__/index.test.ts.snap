// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tailwind4-vite > applying to an ember app > works via CLI 1`] = `""`;

exports[`tailwind4-vite > applying to an ember app > works via CLI 2`] = `""`;
